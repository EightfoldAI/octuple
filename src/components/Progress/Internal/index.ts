import OcLine from './OcLine';
import OcCircle from './OcCircle';

export type { OcProgressProps } from './OcProgress.types';
export { OcCircle, OcLine };
export default { OcCircle, OcLine };
