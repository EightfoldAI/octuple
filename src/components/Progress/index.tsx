import Progress from './Progress';

export { ProgressProps, ProgressSize } from './Progress.types';

export default Progress;
