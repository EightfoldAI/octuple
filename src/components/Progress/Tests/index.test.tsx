import React from 'react';
import Progress from '..';
import type { ProgressProps } from '..';
import ProgressSteps from '../Steps';
import { handleGradient, sortGradient } from '../Line';
import { render } from '../../../tests/Utilities';
import { ProgressVariant } from '../Progress.types';

describe('Progress', () => {
  test('successPercent should decide the progress status when it exists', () => {
    const { container: wrapper, rerender } = render(
      <Progress percent={100} success={{ percent: 50 }} />
    );
    expect(wrapper.querySelectorAll('.progress-status-success')).toHaveLength(
      0
    );

    rerender(<Progress percent={50} success={{ percent: 100 }} />);
    expect(wrapper.querySelectorAll('.progress-status-success')).toHaveLength(
      1
    );

    rerender(<Progress percent={100} success={{ percent: 0 }} />);
    expect(wrapper.querySelectorAll('.progress-status-success')).toHaveLength(
      0
    );
  });

  test('render out-of-range progress', () => {
    const { container: wrapper } = render(<Progress percent={120} />);
    expect(wrapper.firstChild).toMatchSnapshot();
  });

  test('render out-of-range progress with info', () => {
    const { container: wrapper } = render(
      <Progress percent={120} showLabels />
    );
    expect(wrapper.firstChild).toMatchSnapshot();
  });

  test('render negative progress', () => {
    const { container: wrapper } = render(<Progress percent={-20} />);
    expect(wrapper.firstChild).toMatchSnapshot();
  });

  test('render negative successPercent', () => {
    const { container: wrapper } = render(
      <Progress percent={50} success={{ percent: -20 }} />
    );
    expect(wrapper.firstChild).toMatchSnapshot();
  });

  test('render format', () => {
    const { container: wrapper } = render(
      <Progress
        percent={50}
        success={{ percent: 10 }}
        format={(percent, successPercent) => `${percent} ${successPercent}`}
      />
    );
    expect(wrapper.firstChild).toMatchSnapshot();
  });

  test('render strokeColor', () => {
    const { container: wrapper, rerender } = render(
      <Progress type="circle" percent={50} strokeColor="red" />
    );
    expect(wrapper.firstChild).toMatchSnapshot();
    rerender(
      <Progress
        strokeColor={{
          from: '#108ee9',
          to: '#87d068',
        }}
        percent={50}
        type="line"
      />
    );
    expect(wrapper.firstChild).toMatchSnapshot();
    rerender(
      <Progress
        strokeColor={{
          '0%': '#108ee9',
          '100%': '#87d068',
        }}
        percent={50}
        type="line"
      />
    );
    expect(wrapper.firstChild).toMatchSnapshot();
  });

  test('render normal progress', () => {
    const { container: wrapper } = render(<Progress status="normal" />);
    expect(wrapper.firstChild).toMatchSnapshot();
  });

  test('render trailColor progress', () => {
    const { container: wrapper } = render(
      <Progress status="normal" trailColor="#ffffff" />
    );
    expect(wrapper.firstChild).toMatchSnapshot();
  });

  test('render successColor progress', () => {
    const { container: wrapper } = render(
      <Progress
        percent={60}
        success={{ percent: 30, strokeColor: '#ffffff' }}
      />
    );
    expect(wrapper.firstChild).toMatchSnapshot();
  });

  test('render successColor progress type="circle"', () => {
    const { container: wrapper } = render(
      <Progress
        percent={60}
        type="circle"
        success={{ percent: 30, strokeColor: '#ffffff' }}
      />
    );
    expect(wrapper.firstChild).toMatchSnapshot();
  });

  test('render successColor progress type="dashboard"', () => {
    const { container: wrapper } = render(
      <Progress
        percent={60}
        type="dashboard"
        success={{ percent: 30, strokeColor: '#ffffff' }}
      />
    );
    expect(wrapper.firstChild).toMatchSnapshot();
  });

  test('render dashboard zero gapDegree', () => {
    const { container: wrapper } = render(
      <Progress type="dashboard" gapDegree={0} />
    );
    expect(wrapper.firstChild).toMatchSnapshot();
  });

  test('render dashboard 295 gapDegree', () => {
    const { container: wrapper } = render(
      <Progress type="dashboard" gapDegree={295} />
    );
    expect(wrapper.firstChild).toMatchSnapshot();
  });

  test('render dashboard 296 gapDegree', () => {
    const { container: wrapper } = render(
      <Progress type="dashboard" gapDegree={296} />
    );
    expect(wrapper.firstChild).toMatchSnapshot();
  });

  test('get correct line-gradient', () => {
    expect(handleGradient({ from: 'test', to: 'test' }).backgroundImage).toBe(
      'linear-gradient(to right, test, test)'
    );
    expect(handleGradient({}).backgroundImage).toBe(
      'linear-gradient(to right, var(--progress-complete-background-color), var(--progress-complete-background-color))'
    );
    expect(
      handleGradient({ from: 'test', to: 'test', '0%': 'test' }).backgroundImage
    ).toBe('linear-gradient(to right, test 0%)');
  });

  test('sort gradients correctly', () => {
    expect(
      sortGradient({ '10%': 'test10', '30%': 'test30', '20%': 'test20' })
    ).toBe('test10 10%, test20 20%, test30 30%');
    expect(
      sortGradient({
        '10%': 'test10',
        '30%': 'test30',
        '20%': 'test20',
        dummy: 'test',
      })
    ).toBe('test10 10%, test20 20%, test30 30%');
  });

  test('should show success status when percent is 100', () => {
    const { container: wrapper } = render(<Progress percent={100} />);
    expect(wrapper.querySelectorAll('.progress-status-success')).toHaveLength(
      1
    );
  });

  test('should show success status when percent is 100 and status is undefined', () => {
    const { container: wrapper } = render(
      <Progress percent={100} status={undefined} />
    );
    expect(wrapper.querySelectorAll('.progress-status-success')).toHaveLength(
      1
    );
  });

  test('should show success status when status is invalid', () => {
    const errorSpy = jest.spyOn(console, 'error').mockImplementation(() => {});
    const { container: wrapper } = render(
      <Progress percent={100} status={'invalid' as ProgressProps['status']} />
    );
    expect(wrapper.querySelectorAll('.progress-status-success')).toHaveLength(
      1
    );
    errorSpy.mockRestore();
  });

  test('should support pill variant', () => {
    const { container: wrapper } = render(
      <Progress percent={20} variant={ProgressVariant.Pill} />
    );
    expect(
      wrapper.querySelector<HTMLDivElement>('.progress-outer-pill')
    ).toBeTruthy();
    expect(wrapper.firstChild).toMatchSnapshot();
  });

  test('should support steps', () => {
    const { container: wrapper } = render(<Progress steps={3} />);
    expect(wrapper.firstChild).toMatchSnapshot();
  });

  test('steps should be changable', () => {
    const { container: wrapper, rerender } = render(
      <Progress steps={5} percent={60} />
    );
    expect(wrapper.querySelectorAll('.progress-steps-item-active').length).toBe(
      3
    );
    rerender(<Progress steps={5} percent={40} />);
    expect(wrapper.querySelectorAll('.progress-steps-item-active').length).toBe(
      2
    );
  });

  test('steps should be changable when has strokeColor', () => {
    const { container: wrapper, rerender } = render(
      <Progress steps={5} percent={60} strokeColor="#1890ff" />
    );
    expect(
      wrapper.querySelectorAll<HTMLDivElement>('.progress-steps-item')[0].style
        .background
    ).toBe('rgb(24, 144, 255)');
    rerender(<Progress steps={5} percent={40} strokeColor="#1890ff" />);
    expect(
      wrapper.querySelectorAll<HTMLDivElement>('.progress-steps-item')[2].style
        .background
    ).toBe('');
    expect(
      wrapper.querySelectorAll<HTMLDivElement>('.progress-steps-item')[1].style
        .background
    ).toBe('rgb(24, 144, 255)');
  });

  test('steps should support trailColor', () => {
    const { container: wrapper } = render(
      <Progress steps={5} percent={20} trailColor="#1890ee" />
    );
    expect(
      wrapper.querySelectorAll<HTMLDivElement>('.progress-steps-item')[1].style
        .background
    ).toBe('rgb(24, 144, 238)');
  });

  test('steps should support border', () => {
    const { container: wrapper } = render(
      <Progress steps={5} percent={20} trailColor="#1890ee" bordered />
    );
    expect(wrapper.querySelector<HTMLDivElement>('.bordered')).toBeTruthy();
  });

  test('steps should support no border', () => {
    const { container: wrapper } = render(
      <Progress steps={5} percent={20} trailColor="#1890ee" bordered={false} />
    );
    expect(wrapper.querySelector<HTMLDivElement>('.bordered')).toBeFalsy();
  });

  test('steps should support pill variant', () => {
    const { container: wrapper } = render(
      <Progress
        steps={5}
        percent={20}
        trailColor="#1890ee"
        variant={ProgressVariant.Pill}
      />
    );
    expect(
      wrapper.querySelector<HTMLDivElement>('.progress-steps-outer-pill')
    ).toBeTruthy();
    expect(wrapper.firstChild).toMatchSnapshot();
  });

  test('should display correct step', () => {
    const { container: wrapper, rerender } = render(
      <Progress steps={9} percent={22.22} />
    );
    expect(wrapper.querySelectorAll('.progress-steps-item-active').length).toBe(
      2
    );
    rerender(<Progress steps={9} percent={33.33} />);
    expect(wrapper.querySelectorAll('.progress-steps-item-active').length).toBe(
      3
    );
    rerender(<Progress steps={9} percent={44.44} />);
    expect(wrapper.querySelectorAll('.progress-steps-item-active').length).toBe(
      4
    );
  });

  test('steps should have default percent 0', () => {
    const { container } = render(<ProgressSteps steps={0} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  describe('Hooks', () => {
    test('"Rendered more hooks than during the previous render"', () => {
      expect(() => {
        const { rerender } = render(
          <Progress percent={60} success={{ percent: 0 }} type="circle" />
        );
        rerender(
          <Progress percent={60} success={{ percent: 10 }} type="circle" />
        );
      }).not.toThrow();
    });
  });
});
