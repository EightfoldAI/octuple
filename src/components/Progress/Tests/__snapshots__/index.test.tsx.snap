// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress render dashboard 295 gapDegree 1`] = `
<div
  class="progress progress-circle progress-status-normal"
>
  <div
    class="bordered"
  >
    <div
      class="progress-inner"
      style="width: 120px; height: 120px; font-size: 22px;"
    >
      <svg
        class="progress-circle"
        viewBox="0 0 100 100"
      >
        <circle
          class="progress-circle-trail"
          cx="50"
          cy="50"
          r="50"
          stroke-linecap="butt"
          stroke-width="62.5"
          style="stroke-dasharray: 56.72320068981571px 314.1592653589793; stroke-dashoffset: 0; transform: rotate(237.5deg); transform-origin: 50% 50%; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s, opacity .3s ease 0s; fill-opacity: 0;"
        />
        <circle
          class="progress-circle-path"
          cx="50"
          cy="50"
          opacity="0"
          r="50"
          stroke-linecap="butt"
          stroke-width="62.5"
          style="stroke-dasharray: 56.72320068981571px 314.1592653589793; stroke-dashoffset: 56.72320068981571; transform: rotate(237.5deg); transform-origin: 50% 50%; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s, opacity .3s ease 0s; fill-opacity: 0; transition-duration: 0s, 0s;"
        />
        <circle
          class="progress-circle-path"
          cx="50"
          cy="50"
          opacity="0"
          r="50"
          stroke-linecap="butt"
          stroke-width="62.5"
          style="stroke: var(--progress-success-background-color); stroke-dasharray: 56.72320068981571px 314.1592653589793; stroke-dashoffset: 56.72320068981571; transform: rotate(237.5deg); transform-origin: 50% 50%; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s, opacity .3s ease 0s; fill-opacity: 0; transition-duration: 0s, 0s;"
        />
      </svg>
    </div>
  </div>
  <span
    class="progress-text"
    title="0%"
  >
    0%
  </span>
</div>
`;

exports[`Progress render dashboard 296 gapDegree 1`] = `
<div
  class="progress progress-circle progress-status-normal"
>
  <div
    class="bordered"
  >
    <div
      class="progress-inner"
      style="width: 120px; height: 120px; font-size: 22px;"
    >
      <svg
        class="progress-circle"
        viewBox="0 0 100 100"
      >
        <circle
          class="progress-circle-trail"
          cx="50"
          cy="50"
          r="50"
          stroke-linecap="butt"
          stroke-width="62.5"
          style="stroke-dasharray: 55.85053606381855px 314.1592653589793; stroke-dashoffset: 0; transform: rotate(238deg); transform-origin: 50% 50%; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s, opacity .3s ease 0s; fill-opacity: 0;"
        />
        <circle
          class="progress-circle-path"
          cx="50"
          cy="50"
          opacity="0"
          r="50"
          stroke-linecap="butt"
          stroke-width="62.5"
          style="stroke-dasharray: 55.85053606381855px 314.1592653589793; stroke-dashoffset: 55.85053606381855; transform: rotate(238deg); transform-origin: 50% 50%; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s, opacity .3s ease 0s; fill-opacity: 0; transition-duration: 0s, 0s;"
        />
        <circle
          class="progress-circle-path"
          cx="50"
          cy="50"
          opacity="0"
          r="50"
          stroke-linecap="butt"
          stroke-width="62.5"
          style="stroke: var(--progress-success-background-color); stroke-dasharray: 55.85053606381855px 314.1592653589793; stroke-dashoffset: 55.85053606381855; transform: rotate(238deg); transform-origin: 50% 50%; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s, opacity .3s ease 0s; fill-opacity: 0; transition-duration: 0s, 0s;"
        />
      </svg>
    </div>
  </div>
  <span
    class="progress-text"
    title="0%"
  >
    0%
  </span>
</div>
`;

exports[`Progress render dashboard zero gapDegree 1`] = `
<div
  class="progress progress-circle progress-status-normal"
>
  <div
    class="bordered"
  >
    <div
      class="progress-inner"
      style="width: 120px; height: 120px; font-size: 22px;"
    >
      <svg
        class="progress-circle"
        viewBox="0 0 100 100"
      >
        <circle
          class="progress-circle-trail"
          cx="50"
          cy="50"
          r="50"
          stroke-linecap="butt"
          stroke-width="62.5"
          style="stroke-dasharray: 314.1592653589793px 314.1592653589793; stroke-dashoffset: 0; transform: rotate(-90deg); transform-origin: 50% 50%; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s, opacity .3s ease 0s; fill-opacity: 0;"
        />
        <circle
          class="progress-circle-path"
          cx="50"
          cy="50"
          opacity="0"
          r="50"
          stroke-linecap="butt"
          stroke-width="62.5"
          style="stroke-dasharray: 314.1592653589793px 314.1592653589793; stroke-dashoffset: 314.1592653589793; transform: rotate(-90deg); transform-origin: 50% 50%; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s, opacity .3s ease 0s; fill-opacity: 0; transition-duration: 0s, 0s;"
        />
        <circle
          class="progress-circle-path"
          cx="50"
          cy="50"
          opacity="0"
          r="50"
          stroke-linecap="butt"
          stroke-width="62.5"
          style="stroke: var(--progress-success-background-color); stroke-dasharray: 314.1592653589793px 314.1592653589793; stroke-dashoffset: 314.1592653589793; transform: rotate(-90deg); transform-origin: 50% 50%; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s, opacity .3s ease 0s; fill-opacity: 0; transition-duration: 0s, 0s;"
        />
      </svg>
    </div>
  </div>
  <span
    class="progress-text"
    title="0%"
  >
    0%
  </span>
</div>
`;

exports[`Progress render format 1`] = `
<div
  class="progress progress-line progress-status-normal"
>
  <div
    class="progress-outer"
  >
    <div
      class="progress-inner bordered"
    >
      <div
        class="progress-bg"
        style="width: 50%; height: 6px;"
      />
      <div
        class="progress-success-bg"
        style="width: 10%; height: 6px;"
      />
    </div>
  </div>
  <div
    class="progress-text-wrapper"
  >
    <span
      class="extremity-label min-label progress-text"
    >
      50 10
    </span>
    <span
      class="extremity-label max-label progress-text"
    >
      100%
    </span>
    <span
      class="value-label progress-text"
    />
    <span
      class="value-label progress-text"
    />
  </div>
</div>
`;

exports[`Progress render negative progress 1`] = `
<div
  class="progress progress-line progress-status-normal"
>
  <div
    class="progress-outer"
  >
    <div
      class="progress-inner bordered"
    >
      <div
        class="progress-bg"
        style="width: 0%; height: 6px;"
      />
    </div>
  </div>
  <div
    class="progress-text-wrapper"
  >
    <span
      class="extremity-label min-label progress-text"
    >
      0%
    </span>
    <span
      class="extremity-label max-label progress-text"
    >
      100%
    </span>
    <span
      class="value-label progress-text"
    />
    <span
      class="value-label progress-text"
    />
  </div>
</div>
`;

exports[`Progress render negative successPercent 1`] = `
<div
  class="progress progress-line progress-status-normal"
>
  <div
    class="progress-outer"
  >
    <div
      class="progress-inner bordered"
    >
      <div
        class="progress-bg"
        style="width: 50%; height: 6px;"
      />
      <div
        class="progress-success-bg"
        style="width: 0%; height: 6px;"
      />
    </div>
  </div>
  <div
    class="progress-text-wrapper"
  >
    <span
      class="extremity-label min-label progress-text"
    >
      50%
    </span>
    <span
      class="extremity-label max-label progress-text"
    >
      100%
    </span>
    <span
      class="value-label progress-text"
    />
    <span
      class="value-label progress-text"
    />
  </div>
</div>
`;

exports[`Progress render normal progress 1`] = `
<div
  class="progress progress-line progress-status-normal"
>
  <div
    class="progress-outer"
  >
    <div
      class="progress-inner bordered"
    >
      <div
        class="progress-bg"
        style="width: 0%; height: 6px;"
      />
    </div>
  </div>
  <div
    class="progress-text-wrapper"
  >
    <span
      class="extremity-label min-label progress-text"
    >
      0%
    </span>
    <span
      class="extremity-label max-label progress-text"
    >
      100%
    </span>
    <span
      class="value-label progress-text"
    />
    <span
      class="value-label progress-text"
    />
  </div>
</div>
`;

exports[`Progress render out-of-range progress 1`] = `
<div
  class="progress progress-line progress-status-success"
>
  <div
    class="progress-outer"
  >
    <div
      class="progress-inner bordered"
    >
      <div
        class="progress-bg"
        style="width: 100%; height: 6px;"
      />
    </div>
  </div>
  <div
    class="progress-text-wrapper"
  >
    <span
      class="extremity-label min-label progress-text"
    >
      <span
        aria-hidden="false"
        class="progress-icon icon-wrapper"
        role="presentation"
      >
        <svg
          role="presentation"
          style="width: 16px; height: 16px;"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.5 2 2 6.5 2 12S6.5 22 12 22 22 17.5 22 12 17.5 2 12 2M12 20C7.59 20 4 16.41 4 12S7.59 4 12 4 20 7.59 20 12 16.41 20 12 20M16.59 7.58L10 14.17L7.41 11.59L6 13L10 17L18 9L16.59 7.58Z"
            style="fill: var(--progress-success-text-color);"
          />
        </svg>
      </span>
    </span>
    <span
      class="extremity-label max-label progress-text"
    >
      100%
    </span>
    <span
      class="value-label progress-text"
    />
    <span
      class="value-label progress-text"
    />
  </div>
</div>
`;

exports[`Progress render out-of-range progress with info 1`] = `
<div
  class="progress progress-line progress-status-success"
>
  <div
    class="progress-outer"
  >
    <div
      class="progress-inner bordered"
    >
      <div
        class="progress-bg"
        style="width: 100%; height: 6px;"
      />
    </div>
  </div>
  <div
    class="progress-text-wrapper"
  >
    <span
      class="extremity-label min-label progress-text"
    >
      <span
        aria-hidden="false"
        class="progress-icon icon-wrapper"
        role="presentation"
      >
        <svg
          role="presentation"
          style="width: 16px; height: 16px;"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.5 2 2 6.5 2 12S6.5 22 12 22 22 17.5 22 12 17.5 2 12 2M12 20C7.59 20 4 16.41 4 12S7.59 4 12 4 20 7.59 20 12 16.41 20 12 20M16.59 7.58L10 14.17L7.41 11.59L6 13L10 17L18 9L16.59 7.58Z"
            style="fill: var(--progress-success-text-color);"
          />
        </svg>
      </span>
    </span>
    <span
      class="extremity-label max-label progress-text"
    >
      100%
    </span>
    <span
      class="value-label progress-text"
    />
    <span
      class="value-label progress-text"
    />
  </div>
</div>
`;

exports[`Progress render strokeColor 1`] = `
<div
  class="progress progress-circle progress-status-normal"
>
  <div
    class="bordered"
  >
    <div
      class="progress-inner"
      style="width: 120px; height: 120px; font-size: 22px;"
    >
      <svg
        class="progress-circle"
        viewBox="0 0 100 100"
      >
        <circle
          class="progress-circle-trail"
          cx="50"
          cy="50"
          r="50"
          stroke-linecap="butt"
          stroke-width="62.5"
          style="stroke-dasharray: 314.1592653589793px 314.1592653589793; stroke-dashoffset: 0; transform: rotate(-90deg); transform-origin: 50% 50%; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s, opacity .3s ease 0s; fill-opacity: 0;"
        />
        <circle
          class="progress-circle-path"
          cx="50"
          cy="50"
          opacity="1"
          r="50"
          stroke-linecap="butt"
          stroke-width="62.5"
          style="stroke: red; stroke-dasharray: 314.1592653589793px 314.1592653589793; stroke-dashoffset: 157.07963267948966; transform: rotate(-90deg); transform-origin: 50% 50%; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s, opacity .3s ease 0s; fill-opacity: 0; transition-duration: 0s, 0s;"
        />
        <circle
          class="progress-circle-path"
          cx="50"
          cy="50"
          opacity="0"
          r="50"
          stroke-linecap="butt"
          stroke-width="62.5"
          style="stroke: var(--progress-success-background-color); stroke-dasharray: 314.1592653589793px 314.1592653589793; stroke-dashoffset: 314.1592653589793; transform: rotate(-90deg); transform-origin: 50% 50%; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s, opacity .3s ease 0s; fill-opacity: 0; transition-duration: 0s, 0s;"
        />
      </svg>
    </div>
  </div>
  <span
    class="progress-text"
    title="50%"
  >
    50%
  </span>
</div>
`;

exports[`Progress render strokeColor 2`] = `
<div
  class="progress progress-line progress-status-normal"
>
  <div
    class="progress-outer"
  >
    <div
      class="progress-inner bordered"
    >
      <div
        class="progress-bg"
        style="width: 50%; height: 6px;"
      />
    </div>
  </div>
  <div
    class="progress-text-wrapper"
  >
    <span
      class="extremity-label min-label progress-text"
    >
      50%
    </span>
    <span
      class="extremity-label max-label progress-text"
    >
      100%
    </span>
    <span
      class="value-label progress-text"
    />
    <span
      class="value-label progress-text"
    />
  </div>
</div>
`;

exports[`Progress render strokeColor 3`] = `
<div
  class="progress progress-line progress-status-normal"
>
  <div
    class="progress-outer"
  >
    <div
      class="progress-inner bordered"
    >
      <div
        class="progress-bg"
        style="width: 50%; height: 6px;"
      />
    </div>
  </div>
  <div
    class="progress-text-wrapper"
  >
    <span
      class="extremity-label min-label progress-text"
    >
      50%
    </span>
    <span
      class="extremity-label max-label progress-text"
    >
      100%
    </span>
    <span
      class="value-label progress-text"
    />
    <span
      class="value-label progress-text"
    />
  </div>
</div>
`;

exports[`Progress render successColor progress 1`] = `
<div
  class="progress progress-line progress-status-normal"
>
  <div
    class="progress-outer"
  >
    <div
      class="progress-inner bordered"
    >
      <div
        class="progress-bg"
        style="width: 60%; height: 6px;"
      />
      <div
        class="progress-success-bg"
        style="width: 30%; height: 6px; background: rgb(255, 255, 255);"
      />
    </div>
  </div>
  <div
    class="progress-text-wrapper"
  >
    <span
      class="extremity-label min-label progress-text"
    >
      60%
    </span>
    <span
      class="extremity-label max-label progress-text"
    >
      100%
    </span>
    <span
      class="value-label progress-text"
    />
    <span
      class="value-label progress-text"
    />
  </div>
</div>
`;

exports[`Progress render successColor progress type="circle" 1`] = `
<div
  class="progress progress-circle progress-status-normal"
>
  <div
    class="bordered"
  >
    <div
      class="progress-inner"
      style="width: 120px; height: 120px; font-size: 22px;"
    >
      <svg
        class="progress-circle"
        viewBox="0 0 100 100"
      >
        <circle
          class="progress-circle-trail"
          cx="50"
          cy="50"
          r="50"
          stroke-linecap="butt"
          stroke-width="62.5"
          style="stroke-dasharray: 314.1592653589793px 314.1592653589793; stroke-dashoffset: 0; transform: rotate(-90deg); transform-origin: 50% 50%; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s, opacity .3s ease 0s; fill-opacity: 0;"
        />
        <circle
          class="progress-circle-path"
          cx="50"
          cy="50"
          opacity="1"
          r="50"
          stroke-linecap="butt"
          stroke-width="62.5"
          style="stroke-dasharray: 314.1592653589793px 314.1592653589793; stroke-dashoffset: 219.9114857512855; transform: rotate(18deg); transform-origin: 50% 50%; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s, opacity .3s ease 0s; fill-opacity: 0; transition-duration: 0s, 0s;"
        />
        <circle
          class="progress-circle-path"
          cx="50"
          cy="50"
          opacity="1"
          r="50"
          stroke-linecap="butt"
          stroke-width="62.5"
          style="stroke: #ffffff; stroke-dasharray: 314.1592653589793px 314.1592653589793; stroke-dashoffset: 219.9114857512855; transform: rotate(-90deg); transform-origin: 50% 50%; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s, opacity .3s ease 0s; fill-opacity: 0; transition-duration: 0s, 0s;"
        />
      </svg>
    </div>
  </div>
  <span
    class="progress-text"
    title="60%"
  >
    60%
  </span>
</div>
`;

exports[`Progress render successColor progress type="dashboard" 1`] = `
<div
  class="progress progress-circle progress-status-normal"
>
  <div
    class="bordered"
  >
    <div
      class="progress-inner"
      style="width: 120px; height: 120px; font-size: 22px;"
    >
      <svg
        class="progress-circle"
        viewBox="0 0 100 100"
      >
        <circle
          class="progress-circle-trail"
          cx="50"
          cy="50"
          r="50"
          stroke-linecap="butt"
          stroke-width="62.5"
          style="stroke-dasharray: 248.70941840919195px 314.1592653589793; stroke-dashoffset: 0; transform: rotate(127.5deg); transform-origin: 50% 50%; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s, opacity .3s ease 0s; fill-opacity: 0;"
        />
        <circle
          class="progress-circle-path"
          cx="50"
          cy="50"
          opacity="1"
          r="50"
          stroke-linecap="butt"
          stroke-width="62.5"
          style="stroke-dasharray: 248.70941840919195px 314.1592653589793; stroke-dashoffset: 174.09659288643437; transform: rotate(213deg); transform-origin: 50% 50%; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s, opacity .3s ease 0s; fill-opacity: 0; transition-duration: 0s, 0s;"
        />
        <circle
          class="progress-circle-path"
          cx="50"
          cy="50"
          opacity="1"
          r="50"
          stroke-linecap="butt"
          stroke-width="62.5"
          style="stroke: #ffffff; stroke-dasharray: 248.70941840919195px 314.1592653589793; stroke-dashoffset: 174.09659288643437; transform: rotate(127.5deg); transform-origin: 50% 50%; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s, opacity .3s ease 0s; fill-opacity: 0; transition-duration: 0s, 0s;"
        />
      </svg>
    </div>
  </div>
  <span
    class="progress-text"
    title="60%"
  >
    60%
  </span>
</div>
`;

exports[`Progress render trailColor progress 1`] = `
<div
  class="progress progress-line progress-status-normal"
>
  <div
    class="progress-outer"
  >
    <div
      class="progress-inner bordered"
      style="background: rgb(255, 255, 255);"
    >
      <div
        class="progress-bg"
        style="width: 0%; height: 6px;"
      />
    </div>
  </div>
  <div
    class="progress-text-wrapper"
  >
    <span
      class="extremity-label min-label progress-text"
    >
      0%
    </span>
    <span
      class="extremity-label max-label progress-text"
    >
      100%
    </span>
    <span
      class="value-label progress-text"
    />
    <span
      class="value-label progress-text"
    />
  </div>
</div>
`;

exports[`Progress should support pill variant 1`] = `
<div
  class="progress progress-line progress-status-normal"
>
  <div
    class="progress-outer progress-outer-pill"
  >
    <div
      class="progress-inner bordered"
    >
      <div
        class="progress-bg"
        style="width: 20%; height: 6px;"
      />
    </div>
  </div>
  <div
    class="progress-text-wrapper"
  >
    <span
      class="extremity-label min-label progress-text"
    >
      20%
    </span>
    <span
      class="extremity-label max-label progress-text"
    >
      100%
    </span>
    <span
      class="value-label progress-text"
    />
    <span
      class="value-label progress-text"
    />
  </div>
</div>
`;

exports[`Progress should support steps 1`] = `
<div
  class="progress progress-steps progress-status-normal"
>
  <div
    class="progress-steps-outer"
  >
    <div
      class="progress-steps-item bordered"
      style="width: 0px; height: 6px;"
    />
    <div
      class="progress-steps-item bordered"
      style="width: 0px; height: 6px;"
    />
    <div
      class="progress-steps-item bordered"
      style="width: 0px; height: 6px;"
    />
  </div>
  <div
    class="progress-text-wrapper"
  >
    <span
      class="extremity-label min-label progress-text"
    >
      0%
    </span>
    <span
      class="extremity-label max-label progress-text"
    >
      100%
    </span>
    <span
      class="value-label progress-text"
    />
    <span
      class="value-label progress-text"
    />
  </div>
</div>
`;

exports[`Progress steps should have default percent 0 1`] = `
<div
  class="progress-steps-outer"
/>
`;

exports[`Progress steps should support pill variant 1`] = `
<div
  class="progress progress-steps progress-status-normal"
>
  <div
    class="progress-steps-outer progress-steps-outer-pill"
  >
    <div
      class="progress-steps-item bordered progress-steps-item-active"
      style="width: 0px; height: 6px;"
    />
    <div
      class="progress-steps-item bordered"
      style="background: rgb(24, 144, 238); border-color: #1890ee; width: 0px; height: 6px;"
    />
    <div
      class="progress-steps-item bordered"
      style="background: rgb(24, 144, 238); border-color: #1890ee; width: 0px; height: 6px;"
    />
    <div
      class="progress-steps-item bordered"
      style="background: rgb(24, 144, 238); border-color: #1890ee; width: 0px; height: 6px;"
    />
    <div
      class="progress-steps-item bordered"
      style="background: rgb(24, 144, 238); border-color: #1890ee; width: 0px; height: 6px;"
    />
  </div>
  <div
    class="progress-text-wrapper"
  >
    <span
      class="extremity-label min-label progress-text"
    >
      20%
    </span>
    <span
      class="extremity-label max-label progress-text"
    >
      100%
    </span>
    <span
      class="value-label progress-text"
    />
    <span
      class="value-label progress-text"
    />
  </div>
</div>
`;
